lib/class-wp-rest-sidebars-controller.php

```diff
@@ -0,0 +1,342 @@
+<?php
+
+/**
+ * An extension for the WP REST API that exposes endpoints for sidebars and widgets.
+ *
+ * PHP version 5.4.0
+ *
+ * Copyright (C) 2015  Martin Pettersson
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author    Martin Pettersson <martin_pettersson@outlook.com>
+ * @copyright 2015 Martin Pettersson
+ * @license   GPLv2
+ * @link      https://github.com/martin-pettersson/wp-rest-api-sidebars
+ */
+
+/**
+ * Class Sidebars_Controller
+ *
+ * @package WP_API_Sidebars\Controllers
+ */
+class WP_REST_Sidebars_Controller extends WP_REST_Controller {
+
+	/**
+	 * Plugins controller constructor.
+	 *
+	 * @since 5.5.0
+	 */
+	public function __construct() {
+		$this->namespace = '__experimental';
+		$this->rest_base = 'sidebars';
+	}
+
+	/**
+	 * Registers the controllers routes
+	 *
+	 * @return null
+	 */
+	public function register_routes() {
+		// lists all sidebars
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base, [
+			[
+				'methods'  => WP_REST_Server::READABLE,
+				'callback' => [ $this, 'get_items' ],
+			],
+		] );
+
+		// lists a single sidebar based on the given id
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<id>[\w-]+)',
+			[
+				[
+					'methods'  => WP_REST_Server::READABLE,
+					'callback' => [ $this, 'get_item' ],
+					'args'     => [
+						'id' => [
+							'description'       => 'The id of a registered sidebar',
+							'type'              => 'string',
+							'validate_callback' => function ( $id ) {
+								return ! is_null( self::get_sidebar( $id ) );
+							},
+						],
+					],
+				],
+			]
+		);
+
+		// lists a single sidebar based on the given id
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<id>[\w-]+)',
+			[
+				[
+					'methods'  => WP_REST_Server::EDITABLE,
+					'callback' => [ $this, 'update_item' ],
+					'args'     => [
+						'id' => [
+							'description'       => 'The id of a registered sidebar',
+							'type'              => 'string',
+							'validate_callback' => function ( $id ) {
+								return ! is_null( self::get_sidebar( $id ) );
+							},
+						],
+					],
+				],
+			]
+		);
+	}
+
+	public function update_item( $request ) {
+		global $wp_registered_widget_updates;
+		$sidebar_id    = $request->get_param( 'id' );
```

Comment:
```suggestion
		$sidebar_id    = $request['id'];
```

Just use the request object directly.